src/boot/memory.c

```diff
@@ -21,17 +22,24 @@
 #endif
 #include "game/puppyprint.h"
 
+#define DOWN(s, align) (((u32)(s)) & ~((align)-1))
+#define DOWN4(s) DOWN(s, 4)
 
 struct MainPoolState {
-    u32 freeSpace;
-    struct MainPoolBlock *listHeadL;
-    struct MainPoolBlock *listHeadR;
-    struct MainPoolState *prev;
+    struct MainPoolContext ctx;
+    struct MainPoolState* prev;
 };
 
-struct MainPoolBlock {
-    struct MainPoolBlock *prev;
-    struct MainPoolBlock *next;
+// Multichar constants AL and AR
+#define MAIN_POOL_FREEABLE_HEADER_MAGIC_LEFT 0x414c
```

Comment:
nit: Not really sure why we're using magic for this instead of an enum value...memory debugging?

Reply:
It is just a canary, it can be anything. Using enum for just 1 value sounds useless.

Reply:
It'd be 2 (left and right), right? Canary checking sounds useful though, so I'll let this slide.